search-app/src/cmr/search/services/query_execution/facets/granule_v2_facets.clj

```diff
@@ -20,3 +22,22 @@
 (def v2-facets-result-field-in-order
   "Defines the v2 facets result field in order"
   [])
+
+(defn single-collection-validation
+  "Validates that the provided query is limited to a single collection. We do this to prevent
+  expensive aggregation queries that would have to run against more than one granule index."
+  [context]
+  (let [collection-ids (:query-collection-ids context)
+        collection-count (count collection-ids)]
+    (when-not (= 1 collection-count)
+      [(str "Granule V2 facets are limited to a single collection, but query matched "
```

Comment:
I think the following would be more readable, and there is no need to call `zero?`.
`[(format "Granule V2 facets are limited to a single collection, but query matched %s collections."
    (if (= 0 collection-count) "an undetermined number of" collection-count))...`

Reply:
Me too. I only used `str` to avoid the long line or 😱  using `(format (str`. I'll go ahead and use `format` and the long line.